File: modified_scripts/rust/bubble_sort/gemma_2b.rs
```rust
fn bubble_sort(nums: &mut [i32]) {
    for i in 0..(nums.len() - 1) {
        for j in 0..(nums.len() - i - 1) {
            if nums[j] > nums[j + 1] {
                swap(&mut nums[j], &mut nums[j + 1]);
            }
        }
    }
}
```

File: modified_scripts/scala/bubble_sort/gemma_2b.scala

def bubbleSort(arr: Array[Int]): Unit = {
  var swapped = false
  for (i <- 0 to arr.length - 1) {
    if (arr(i) > arr(i + 1)) {
      swap(arr, i, i + 1)
      swapped = true
    }
  }
}

private def swap(arr: Array[Int], i: Int, j: Int): Unit = {
  val temp = arr(i)
  arr(i) = arr(j)
  arr(j) = temp
}
